src/main/java/seedu/address/logic/parser/wallet/WalletFindCommandParser.java

```diff
@@ -20,14 +31,38 @@
      * @throws ParseException if the user input does not conform the expected format
      */
     public WalletFindCommand parse(String args) throws ParseException {
+
         String trimmedArgs = args.trim();
         if (trimmedArgs.isEmpty()) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, WalletFindCommand.MESSAGE_USAGE));
         }
 
-        String[] nameKeywords = trimmedArgs.split("\\s+");
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_AMOUNT, PREFIX_DATE, PREFIX_TAG);
+
+        WalletPredicate walletFindPredicate = null;
+
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            String[] nameKeywords = argMultimap.getValue(PREFIX_NAME).get().split("\\s+");
+            walletFindPredicate = new DescriptionContainsKeywordsPredicate(Arrays.asList(nameKeywords));
+        } else if (argMultimap.getValue(PREFIX_AMOUNT).isPresent()) {
+            String[] nameKeywords = argMultimap.getValue(PREFIX_AMOUNT).get().split("\\s+");
+            walletFindPredicate = new AmountContainsKeywordsPredicate(Arrays.asList(nameKeywords));
+        } else if (argMultimap.getValue(PREFIX_DATE).isPresent()) {
```

Comment:
Currently here the user has to enter date 'yyyy-mm-dd' to find transactions. But I think ideally they should be entering the date format they usually enter for other commands which is 'dd/mm/yyyy' so they get less confused,

Reply:
okayy changed!